### PR TITLE
add a new api to get config details (close #1831); add jwt-analyzer (close #1369)

### LINKED ISSUE #1369: Add jwt inspector to hasura console
It would be super nice if we had a little JWT inspector within the console and also a test endpoint to test if the JWT was being verified and accepted by Hasura (to verify hasura configuration). 
Maybe a little inspect icon next to the JWT authorization header if the console detects it?

@rikinsk @karthikvt26 @praveenweb @wawhal thoughts?

### LINKED ISSUE #1831: admin only endpoint to dump configuration settings
Admin only endpoint to dump JWT config. Required for #1369.

Refer to discussion at: https://github.com/hasura/graphql-engine/pull/1770#issuecomment-475136297

### PR DESCRIPTION
**Pull request by shahidhk**
### Description
This PR adds a new admin-only API which returns the current server configuration, mainly auth-related, to be used by the console.

API:
```http
GET /v1alpha1/config
```
response
```json
{
  "version": "v1-query-config-90ebe37b",
  "is_admin_secret_set": true,
  "is_auth_hook_set": false,
  "is_jwt_set": true,
  "jwt": {
    "claims_namespace": "custom_namespace",
    "claims_format": "json"
  }
}
```

Using this API, the PR adds a JWT analyzer to console which helps users to inspect the JWT to see Hasura claims etc.

### Affected components 

- Console
- Server
- Docs

### Related Issues
close #1831, close #1369, #1770 

### Solution and Design
A new API is added to expose these details.

### Steps to test and verify

Checks
- [x] no jwt configured but auth header is set
- [x] valid jwt
- [x] expired jwt
- [x] invalid namespace
- [x] unverified jwt

### Limitations, known bugs & workarounds
~Tests are pending. Will add once the review is done and API is finalised.~
NA

### CODE REVIEW

**Comment by shahidhk** [comment]
The api needs to be changed to use a different URL.

**Comment by shahidhk** [comment]
API is changed to `v1alpah1/config`.

**Comment by karthikvt26** [comment]
Steps to test:

1) Configure GraphQL engine to run with access secret and HASURA_GRAPHQL_JWT_SECRET
2) Generate a JWT token using the secret token configured in the previous step
3) Head to API explorer add a header as follows
    `Authorization: Bearer <token>`
4) Click on the `+` icon to analyze the token

@rikinsk

**Comment by shahidhk** [comment]
@karthikvt26 Can you add a JWT analyzer screenshot to the Description?

**Comment by shahidhk** [comment]
The endpoint is v1alpha1, not v1

File: docs/graphql/manual/api-reference/index.rst [hunk 1]
```diff
@@ -22,7 +22,9 @@ Available APIs
 +-----------------+----------------------------------------+------------------+
 | Health          | :ref:`/healthz <health_api>`           | Public           |
 +-----------------+----------------------------------------+------------------+
-| PG Dump         | :ref:`/v1alpha1/pg_dump <pg_dump_api>` | Admin only       |
+| PG Dump         | :ref:`/v1alpha/pg_dump <pg_dump_api>`  | Admin only       |
```
Comment (shahidhk) [inline docs/graphql/manual/api-reference/index.rst]:
```suggestion
| PG Dump         | :ref:`/v1alpha1/pg_dump <pg_dump_api>`  | Admin only       |
```

File: docs/graphql/manual/api-reference/index.rst [hunk 2]
```diff
@@ -75,12 +77,22 @@ out what the errors are.
 pg_dump API
 ^^^^^^^^^^^
 
-The ``/v1alpha1/pg_dump`` is an admin-only endpoint that can be used to execute ``pg_dump`` on the
-Postgres instance connected to Hasura. The ``pg_dump`` CLI tool's argument can
-be passed as POST request body to the API and the response is sent back to the
-client.
+The ``/v1alpha/pg_dump`` is an admin-only endpoint that can be used to execute
```
Comment (shahidhk) [inline docs/graphql/manual/api-reference/index.rst]:
```suggestion
The ``/v1alpha1/pg_dump`` is an admin-only endpoint that can be used to execute
```

File: docs/graphql/manual/api-reference/pgdump.rst [hunk 3]
```diff
@@ -18,7 +18,7 @@ such as taking data dump etc.
 Endpoint
 --------
 
-All requests are ``POST`` requests to the ``/v1alpha1/pg_dump`` endpoint.
+All requests are ``POST`` requests to the ``/v1alpha/pg_dump`` endpoint.
```
Comment (shahidhk) [inline docs/graphql/manual/api-reference/pgdump.rst]:
```suggestion
All requests are ``POST`` requests to the ``/v1alpha1/pg_dump`` endpoint.
```

File: docs/graphql/manual/api-reference/pgdump.rst [hunk 4]
```diff
@@ -28,7 +28,7 @@ Request
 
 .. code-block:: http
 
-   POST /v1alpha1/pg_dump HTTP/1.1
+   POST /v1alpha/pg_dump HTTP/1.1
```
Comment (shahidhk) [inline docs/graphql/manual/api-reference/pgdump.rst]:
```suggestion
   POST /v1alpha1/pg_dump HTTP/1.1
```

**Pull request merged**